### DISCUSSION TITLE
Request a custom touch layout for Call of Duty Black Ops 6, using the controls from Call of Duty Mobile as a reference.

### DISCUSSION THREAD

**Discussion by Daplugg23** [redphx/better-xcloud#542] [Custom touch controls]
# Custom Touch Controller Layout for Call of Duty Black Ops 6

Userscript to improve Xbox Cloud Gaming (xCloud) and Remote Play experiences on web browsers.

### Left Side of the Screen

- **Virtual Joystick**  
  - **Function:** Controls player movement (forward, backward, left, right).  
  - **Placement:** Bottom-left corner.  
  - **Interaction:** Drag in any direction to move the character accordingly.

- **Sprint Button**  
  - **Function:** Enables the player to sprint.  
  - **Placement:** Above the virtual joystick.  
  - **Interaction:** Tap to toggle sprinting on/off.

### Right Side of the Screen

- **Fire Button**  
  - **Function:** Fires the weapon.  
  - **Placement:** Two fire buttons—one at the top-right corner and another mid-right side for ease of access.  
  - **Interaction:** Tap and hold to fire continuously.

- **Aim Down Sights (ADS) Button**  
  - **Function:** Toggles aiming down the weapon's sights.  
  - **Placement:** Above the bottom-right fire button.  
  - **Interaction:** Tap to toggle ADS on/off.

- **Crouch Button**  
  - **Function:** Makes the player crouch.  
  - **Placement:** Bottom-right corner.  
  - **Interaction:** Tap to crouch; tap again to stand up.

- **Jump Button**  
  - **Function:** Makes the player jump.  
  - **Placement:** Right of the crouch button.  
  - **Interaction:** Tap to jump.

- **Reload Button**  
  - **Function:** Reloads the current weapon.  
  - **Placement:** Above the crouch button.  
  - **Interaction:** Tap to reload.

- **Grenade Button**  
  - **Function:** Throws a grenade or uses tactical equipment.  
  - **Placement:** Left of the jump button.  
  - **Interaction:** Tap to equip, then aim and release to throw.

- **Melee Button**  
  - **Function:** Performs a melee attack.  
  - **Placement:** Left of the grenade button.  
  - **Interaction:** Tap to melee.

- **Special Ability Button**  
  - **Function:** Activates special abilities or scorestreaks.  
  - **Placement:** Above the grenade button.  
  - **Interaction:** Tap to activate.

- **Weapon Switch Button**  
  - **Function:** Switches between primary and secondary weapons.  
  - **Placement:** Above the melee button.  
  - **Interaction:** Tap to switch weapons.

### Center of the Screen

- **Mini-Map**  
  - **Function:** Displays the in-game map.  
  - **Placement:** Top-center.  
  - **Interaction:** Tap to enlarge if applicable.

- **Health Bar and Ammo Count**  
  - **Function:** Shows player's health and remaining ammunition.  
  - **Placement:** Below the mini-map.

### Additional Controls

- **Scope Button**  
  - **Function:** For weapons equipped with scopes (e.g., sniper rifles).  
  - **Placement:** Near the ADS button.  
  - **Interaction:** Tap to look through the scope.

- **Chat Button**  
  - **Function:** Opens the chat interface.  
  - **Placement:** Top-left corner.  
  - **Interaction:** Tap to open chat.

- **Pause/Menu Button**  
  - **Function:** Opens the game menu.  
  - **Placement:** Top-right corner.  
  - **Interaction:** Tap to pause or access settings.

## Summary of Button Functions

- **Movement:** Virtual joystick on the left controls movement.  
- **Sprinting:** Dedicated sprint button above the joystick.  
- **Firing Weapons:** Two fire buttons on the right for flexibility.  
- **Aiming:** ADS button toggles aim down sights.  
- **Jumping and Crouching:** Jump and crouch buttons at the bottom-right corner.  
- **Reloading:** Reload button above the crouch button.  
- **Grenades and Melee:** Grenade and melee buttons situated for quick access.  
- **Weapon Switching:** Easily switch weapons with the weapon switch button.  
- **Special Abilities:** Activate scorestreaks or special skills with the special ability button.  
- **Mini-Map and HUD Elements:** Important information displayed at the top-center.

## Reference Layout

The layout closely mirrors the default control scheme of **Call of Duty Mobile**, ensuring familiarity for players who have experience with the mobile version.

**Comment by Tique0312** [comment]
Any way to increase the size of the buttons and make the right stick work like a mouse? There are problems with movement and shooting

**Comment by gLa3-ark** [comment]
Want this please

**Comment by bah99** [comment]
Hi, I have made a highly customizable virtual controller userscript for fps games https://gist.github.com/bah99/2ab1ac27555089247ed3b0220d1e9ea7
